polling_stations/apps/councils/management/commands/import_councils.py

```diff
@@ -83,97 +87,82 @@ def get_json(self, url):
 
-        # ids of any councils we're going to delete
-        deleteme = []
+    def get_council_name(self, council_data):
+        """
+        At the time of writing, the council name can be NULL in the API
+        meaning we can't rely on the key being populated in all cases.
 
-        for council in councils:
-            if not council.council_id in settings.OLD_TO_NEW_MAP:
-                continue
-            code = settings.OLD_TO_NEW_MAP[council.council_id]
-            if code in new_council_objects:
-                new_council_objects[code].area = union_areas(
-                    new_council_objects[code].area, council.area
-                )
-                self.stdout.write("{} --> {}".format(council.council_id, code))
-                deleteme.append(council.council_id)
-
-        councils = [c for c in councils if c.council_id not in deleteme]
-
-        for code, council in new_council_objects.items():
-            councils.append(council)
-
-        return councils
-
-    def load_contact_details(self, contact_type):
-        files = [
-            "./polling_stations/apps/councils/data/england-wales.csv",
-            "./polling_stations/apps/councils/data/ni.csv",
-        ]
-
-        if contact_type == "vjbs":
-            files.append("./polling_stations/apps/councils/data/scotland-vjbs.csv")
-        if contact_type == "councils":
-            files.append("./polling_stations/apps/councils/data/scotland-councils.csv")
-
-        for filename in files:
-            with open(os.path.abspath(filename)) as infile:
-                reader = csv.reader(infile)
-                Row = namedtuple("Row", next(reader))
-                for row in map(Row._make, reader):
-                    self.contact_details[row.gss] = row
-
-    def attach_contact_details(self, councils, contact_type):
-        self.stdout.write("Attaching contact details...")
-        self.load_contact_details(contact_type)
-        for council in councils:
-            if council.council_id not in self.contact_details:
-                # Skip councils that exist in the database with no contact
-                # details. It's possible older objects exist that we aren't
-                # interested in
-                continue
-            contact_details = self.contact_details[council.council_id]
-            council.name = contact_details.name
-            council.website = contact_details.website
-            council.email = contact_details.email
-            council.phone = contact_details.phone
-            council.address = contact_details.address
-            council.postcode = contact_details.postcode
+        This is normally only an issue with councils covered by EONI, so if
+        we see one of them without a name, we assign a hardcoded name.
+
+        """
+        name = None
+        if council_data["official_name"]:
+            name = council_data["official_name"]
+        else:
+            if council_data["code"] in NIR_IDS:
```

Comment:
I'm inferring from this line (and other stuff in this PR) that Numiko/the Commission are using gov.uk register codes as they key for councils/areas. This is probably the most concerning thing here, especially given the potential upcoming changes to councils. GSS codes have their problems - its annoying that they change in response to minor boundary changes but at least when re-organisations happen we can get hold of the new GSS codes ahead of time. It would be useful if they were actually published ahead of time, but in lieu of that at least ONS Geography support are really helpful (and consistently have been for many years). Gov.uk register codes are conceptually nice but they're a black box, their release schedule is non-existent and their support are somewhere on the spectrum between useless and actively obstructive (did I forward you my last email exchange with them?). Even though they're a conceptually nicer identifier if we care about an organisation rather than a polygon, these are big problems and this sets up yourself/Numiko/the EC for an annoying interoperability/temp codes problem.

Just as a kind of status update on this
https://elections.democracyclub.org.uk/organisations/combined-authority/north-of-tyne/2018-11-02/
This CA has existed since 2018. All other CAs have a register code. This one doesn't. I've never been able to get an answer on whether this is because it isn't supposed to have one (why?) or if its an oversight. It still has a temp slug in EE.

https://elections.democracyclub.org.uk/organisations/local-authority/buckinghamshire/2020-04-01/
This UA was created in April 2020. We gave it a temp slug in EE because it didn't have a register code yet. I was unable to really get an answer on when it would be assigned one. This one actually does have a code now, so I'll raise an issue on EE to update it.

https://elections.democracyclub.org.uk/organisations/local-authority/north-northamptonshire/2020-05-07/
https://elections.democracyclub.org.uk/organisations/local-authority/west-northamptonshire/2020-05-07/
These UAs will come into existence in April 2021 (the start dates in EE are 2020 because they were supposed to have "shadow" elections in 2020). The register codes have not been published yet. We don't know when they will be and we don't know what they are yet (but we do already know the GSS codes).

Obviously its you that has to live with the consequences of this, not me, so I'll make my point and leave you to think about it. Its worth at least having a think about what to do about this stuff/how to do temp IDs anyway..